peekingduck/nodes/output/media_writer.py

```diff
@@ -72,13 +80,26 @@ def run(self, inputs: Dict[str, Any]) -> Dict[str, Any]:
             if self.writer:  # images automatically releases writer
                 self.writer.release()
             return {}
-        if not self._file_name:
-            self._prepare_writer(
-                inputs["filename"], inputs["img"], inputs["saved_video_fps"]
-            )
-        if inputs["filename"] != self._file_name:
+        # init
+        self._input_type = self._detect_media_type(inputs["filename"])
+        if self.output_filename is not None:
+            # Use input_filename extension if output_filename extension is empty
+            if Path(self.output_filename).suffix == "":
+                self.output_filename = "".join(
+                    [self.output_filename, f"{Path(inputs['filename']).suffix}"]
```

Comment:
Above `'filename'` should use double instead of single quotes.

Reply:
My bad: the single quotes are needed within the f-string.
But a better implementation is to use the `PurePath.with_suffix()` api suggested by Yier.

Reply:
No worry. Refactor code to use Pathlib.